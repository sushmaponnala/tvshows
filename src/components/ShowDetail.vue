<template>
  <b-container>
    <b-row class="justify-content-md-center">
      <b-col>
        <div class="card-body" v-if="details.image && details.image.medium">
          <b-card :img-src="details.image.medium" img-alt="Card image" img-left class="mb-3">
            <b-card-text>
              <h4>{{details.name}}</h4>
              <p v-html="details.summary"></p>
              <small>
                Rating:
                <span class="text-muted">{{details.rating.average}}</span>
                <br />Genre(s):
                <span class="text-muted">{{details.genres.join(", ")}}</span>
              </small>
            </b-card-text>
          </b-card>
        </div>
      </b-col>
    </b-row>
  </b-container>
</template>

<script>
import axios from "axios";
const getData = (id, cb) => {
  axios
    .get(`https://api.tvmaze.com/shows/${id}`)
    .then(response => cb(null, response))
    .catch(e => {
      cb(e);
    });
};

export default {
  name: "ShowDetail",
  data() {
    return {
      details: {}
    };
  },
  created() {
    getData(this.$route.params.id, (err, response) => {
      this.details = response.data;
    });
  }
};
</script>
