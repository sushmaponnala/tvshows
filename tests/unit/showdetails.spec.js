import { shallowMount, createLocalVue } from '@vue/test-utils'
import ShowDetail from '@/components/ShowDetail.vue'
import { BootstrapVue } from 'bootstrap-vue'
// import VueRouter from 'vue-router'

const localVue = createLocalVue()

localVue.use(BootstrapVue)

describe('ShowDetail.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(ShowDetail, {
      propsData: {},
      data() {
        return {
          details: {
            name: "Breaking Bad",
            summary: "<p><b>Breaking Bad</b> follows protagonist Walter White, a chemistry teacher who lives in New Mexico with his wife and teenage son who has cerebral palsy. White is diagnosed with Stage III cancer and given a prognosis of two years left to live. With a new sense of fearlessness based on his medical prognosis, and a desire to secure his family's financial security, White chooses to enter a dangerous world of drugs and crime and ascends to power in this world. The series explores how a fatal diagnosis such as White's releases a typical man from the daily concerns and constraints of normal society and follows his transformation from mild family man to a kingpin of the drug trade.</p>",
            image: {
              medium: "http://static.tvmaze.com/uploads/images/medium_portrait/0/2400.jpg"
            },
            genres: [
              "Drama",
              "Crime",
              "Thriller"
            ],
            rating: {
              average: 9.3
            }
          },
        }
      },
      mocks: {
        $route: {
          path: '/showdetail/169',
          params: {
            id: 169
          }
        }
      },
      localVue
    })
    expect(wrapper.element).toMatchSnapshot();
  })
})
