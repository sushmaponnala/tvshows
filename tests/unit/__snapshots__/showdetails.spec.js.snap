// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowDetail.vue renders correctly 1`] = `
<b-container-stub
  tag="div"
>
  <b-row-stub
    class="justify-content-md-center"
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <div
        class="card-body"
      >
        <b-card-stub
          bodytag="div"
          class="mb-3"
          footertag="div"
          headertag="div"
          imgalt="Card image"
          imgleft="true"
          imgsrc="http://static.tvmaze.com/uploads/images/medium_portrait/0/2400.jpg"
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          titletag="h4"
        >
          <b-card-text-stub
            texttag="p"
          >
            <h4>
              Breaking Bad
            </h4>
             
            <p>
              <p>
                <b>
                  Breaking Bad
                </b>
                 follows protagonist Walter White, a chemistry teacher who lives in New Mexico with his wife and teenage son who has cerebral palsy. White is diagnosed with Stage III cancer and given a prognosis of two years left to live. With a new sense of fearlessness based on his medical prognosis, and a desire to secure his family's financial security, White chooses to enter a dangerous world of drugs and crime and ascends to power in this world. The series explores how a fatal diagnosis such as White's releases a typical man from the daily concerns and constraints of normal society and follows his transformation from mild family man to a kingpin of the drug trade.
              </p>
            </p>
             
            <small>
              
              Rating:
              
              <span
                class="text-muted"
              >
                9.3
              </span>
               
              <br />
              Genre(s):
              
              <span
                class="text-muted"
              >
                Drama, Crime, Thriller
              </span>
            </small>
          </b-card-text-stub>
        </b-card-stub>
      </div>
    </b-col-stub>
  </b-row-stub>
</b-container-stub>
`;
